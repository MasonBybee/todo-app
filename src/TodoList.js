import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid";

const TodoList = () => {
  const initialState = [{ todo: "Get Milk", id: uuid() }];
  const [todos, setTodos] = useState(initialState);
  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };
  const addTodo = (newTodo) => {
    setTodos((todos) => [...todos, { ...newTodo, id: uuid() }]);
  };
  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>
        {todos.map(({ todo, id }) => (
          <Todo removeTodo={removeTodo} todo={todo} id={id} key={id} />
        ))}
      </ul>
    </div>
  );
};

export default TodoList;
