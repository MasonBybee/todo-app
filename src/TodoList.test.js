import { fireEvent, render } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", () => {
  render(<TodoList />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", () => {
  const { queryByText, getByLabelText } = render(<TodoList />);
  const value = getByLabelText("Todo:");
  const btn = queryByText("Submit Todo");
  expect(queryByText("Get eggs")).not.toBeInTheDocument();
  fireEvent.change(value, { target: { value: "Get eggs" } });
  fireEvent.click(btn);
  expect(queryByText("Get eggs")).toBeInTheDocument();
});
