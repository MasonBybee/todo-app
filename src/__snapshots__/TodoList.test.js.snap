// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        Todo: 
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Submit Todo
      </button>
    </form>
    <ul>
      <li>
        Get Milk
        <button>
          X
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
